src/components/structures/MatrixChat.js

```diff
@@ -635,23 +657,35 @@ module.exports = React.createClass({
                 action: 'start_post_registration',
             });
         } else if (screen.indexOf('room/') == 0) {
-            var roomString = screen.split('/')[1];
+            var roomString = screen.substring(5);
+            var eventId;
+
+            // extract event id, if one is given
+            var idx = roomString.indexOf('/');
+            if (idx >= 0) {
+                eventId = roomString.substring(idx+1);
+                roomString = roomString.substring(0, idx);
```

Comment:
We should probably handle the case where we get extra paths (more `/` characters). I suggested just using the last path segment, but alternatively we could fail hard or just leave it as is. What do you think?